lib/ramfs/ramfs.h

```diff
@@ -36,29 +36,66 @@
 #include <vfscore/prex.h>
 #include <stdbool.h>
 
-/*
- * File/directory node for RAMFS
+/**
+ * struct ramfs_node - A file/directory node for RamFS
  */
 struct ramfs_node {
-	struct ramfs_node *rn_next;   /* next node in the same directory */
-	struct ramfs_node *rn_child;  /* first child node */
-	int rn_type;    /* file or directory */
-	char *rn_name;    /* name (null-terminated) */
-	size_t rn_namelen;    /* length of name not including terminator */
-	size_t rn_size;    /* file size */
-	char *rn_buf;    /* buffer to the file data */
-	size_t rn_bufsize;    /* allocated buffer size */
+	/* Next node in the same directory */
+	struct ramfs_node *rn_next;
+	/* First child node */
+	struct ramfs_node *rn_child;
+	/* Type: file or directory */
```

Comment:
I would add here the possible values: `VREG`, `VLNK`, `VDIR` and their explanation.